raiden/connection_manager.py

```diff
@@ -313,9 +313,7 @@ def funds_remaining(self):
         """The remaining funds after subtracting the already deposited amounts.
         """
         if self.funds > 0:
-            remaining = self.funds - sum(
-                channel.contract_balance for channel in self.open_channels
-            )
+            remaining = self.funds - (self.sum_deposits - self.previous_deposits)
```

Comment:
I'm not convinced that it makes sense to record the `previous_deposits`, especially in this way, i.e. set the value on each call to `connect`. This just removes the idem-potency from `connect` without providing any separation between previously opened channels and later opened channels. I suggest to remove `previous_deposits`.

Reply:
Imagine the following scenario: someone has a previously joined network, with 400 tokens deposited in 3 channels, and then call join again, trying to join some more channels, but now, with deposit of 10, hoping to join, e.g. 2 more channels with this deposit. I think the expected behavior would be him to end up with 5 channels, and 410 tokens deposited on them. Without this `previous_deposits` logic, the `remaining`value would be negative (`-390`), and  no deposits would be made to the 2 new channels, which would then be left unfunded.

Reply:
But, if thhe expectation above is wrong, and the user should always try to join with the **total** deposit amout (i.e. deposit always greater than current deposited sum), then yes, we don't need it.